oem/ibm/libpldmresponder/oem_ibm_handler.hpp

```diff
@@ -471,6 +512,43 @@ class Handler : public oem_platform::Handler
      */
     void triggerHostEffecter(bool value, std::string path);
 
+    /** @brief Allows/Deny DMA operation based on boot progress state and Host
+     * State. DMA is allowed if BootProgressStage is one of the following
+     * SystemInitComplete
+     * OSRunning
+     * SecondaryProcInit
+     * OSStart
+     * SystemSetup
+     * DMA is also allowed in Boot Progress State Unspecified only if
+     * host state is TransitioningToOff
+     *
+     *  @param[out] bool - true is returned if DMA operation is allowed
+     *                     false means DMA is declined
+     */
+    bool getBootProgressState()
+    {
+        if ((bootProgressState != "xyz.openbmc_project.State.Boot.Progress."
+                                  "ProgressStages.SystemInitComplete") &&
```

Comment:
is the condition && or ||? are we checking if any of these conditions are met or all conditions to be met?

Reply:
All these conditions should be met